numpyro/infer/ensemble_util.py

```diff
@@ -0,0 +1,110 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import warnings
+
+import numpy as np
+
+from jax import lax, vmap
+import jax.numpy as jnp
+
+from jax._src import dtypes
+from jax._src.tree_util import tree_flatten, tree_unflatten
+from jax._src.util import safe_zip, unzip2, HashablePartial
+
+zip = safe_zip
+
+
+def _get_nondiagonal_pairs(n):
```

Comment:
@fehiepsi do you have any suggestions on the utility methods in this file? can any of this functionality be covered by existing utils?